File: day05/test/solution_test.dart
import 'dart:io';
import 'package:test/test.dart';
import '../lib/solution.dart';

void main() {
  late String exampleInput;

  setUpAll(() async {
    exampleInput = await File('example.txt').readAsString();
  });

  group('Day 5', () {
    test('Part 1 - Example', () {
      final result = solvePart1(exampleInput);
      expect(result, equals(3));
    });

    test('Part 2 - Example', () {
      final result = solvePart2(exampleInput);
      expect(result, equals(14));
    });
  });
}

File: day10/solution.test.ts
import { readInput, solvePart1, solvePart2 } from "./solution";

describe("Day 10", () => {
  const exampleInput = readInput("example.txt");

  test("Part 1 - Example", () => {
    const result = solvePart1(exampleInput);
    expect(result).toBe(7);
  });

  test.skip("Part 2 - Example", () => {
    const result = solvePart2(exampleInput);
    expect(result).toBe(0); // Update with expected value
  });
});
